pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
Yes. If users explicitly set --wait=true while deleting namespace then this code waits. If they do not set it explicitly, then it does not wait (to continue existing behavior). Does that sound good?

Sorry am having trouble understanding, what change do you want me to make.